### DISCUSSION TITLE
`tootctl media remove-orphans` removes thousands of orphans per month. Is this expected?

### DISCUSSION THREAD

**Discussion by nadiamoe** [mastodon/mastodon#24430] [Q&A] [answered]
Hello!

I run a small instance with ~100 MAUs, and pretty active federation with the rest of the fediverse. I have set 15d of media cache retention and 720d of content cache retention. I run `tootctl media remove` regularly (once a week).

I have noticed that if I run `tootctl media remove-orphans`, the command gets rid of thousands of files, including avatars, post attachments, preview cards, etc. This happens even if only 1-2 months have passed since I last ran `remove-orphans`.

I would like to ask the community if this is considered normal behavior. I haven't seen much documentation regarding this command, and certainly not seen any recommendations to run it periodically, but [this post](https://mastodon.social/@Gargron/103626182691999440) from Eugen from some time ago hints that this should not be happening, which worries me a bit.

> If your server has been around for a long time, it's likely that you have at least some of those due to various previous bugs and object storage glitches.

Do you run this command frequently? Is it recommended to do so?

**Comment by nadiamoe** [comment, marked as answer]
Quick update on this, since I upped S3 timeouts `remove-orphans` no longer finds any orphan altogether. For my S3 provider (backblaze) I'm using
```
      S3_OPEN_TIMEOUT: 10
      S3_READ_TIMEOUT: 10
```

**Reply by muddyland** [reply]
I was having an issue with timeouts on my MinIO storage, somehow ended up here, and this saved my day! Thanks for sharing!

**Reply by kpfleming** [reply]
Same here!
